src/classes/devices/difluidMicrobalance.ts

```diff
@@ -0,0 +1,172 @@
+export class DifluidMicrobalance extends BluetoothScale {
+  public static DEVICE_NAME = 'difluidmicrobalance';
+  public static SERVICE_UUID = '00EE';
+  public static CHAR_UUID = 'AA01';
+
+  protected override weight: Weight = {
+    actual: 0,
+    old: 0,
+    smoothed: 0,
+    oldSmoothed: 0,
+  };
+
+  private logger: Logger;
+  public scaleUnit = 'g';
+
+  constructor(data: PeripheralData) {
+    super(data);
+    this.logger = new Logger('DiFluid Scale');
+    this.connect();
+  }
+
+  public static test(device: any): boolean {
+    return (
+      device && device.name
+      // && this.DEVICE_NAME.includes(device.name)
+    );
+  }
+
+  public override async connect() {
+    this.logger.log('connecting...');
+    await this.attachNotification();
+    await this.setUnitToGram();
+    await this.enableAutoNotifications();
+  }
+
+  public override async tare() {
+    this.weight.smoothed = 0;
+    this.weight.actual = 0;
+    this.weight.oldSmoothed = 0;
+    this.weight.old = 0;
+    this.setWeight(0);
+
+    await this.write(
+      new Uint8Array([0xdf, 0xdf, 0x03, 0x02, 0x01, 0x01, 0xc5])
+    );
+  }
+
+  public override disconnectTriggered(): void {
+    this.logger.log('Disconnecting...');
+
+    this.deattachNotification();
+  }
+
+  public async setUnitToGram() {
+    await this.write(
+      new Uint8Array([0xdf, 0xdf, 0x01, 0x04, 0x01, 0x00, 0xc4])
+    );
+  }
+
+  public override async setTimer(_timer: SCALE_TIMER_COMMAND) {
+    this.logger.log('Setting Timer command ' + _timer + '...');
+
+    if (_timer === SCALE_TIMER_COMMAND.START) {
+      await this.write(
+        new Uint8Array([0xdf, 0xdf, 0x03, 0x02, 0x01, 0x00, 0xc4])
+      );
+    } else if (_timer === SCALE_TIMER_COMMAND.STOP) {
+      await this.write(
+        new Uint8Array([0xdf, 0xdf, 0x03, 0x01, 0x01, 0x00, 0xc3])
+      );
+      await this.write(
+        new Uint8Array([0xdf, 0xdf, 0x03, 0x02, 0x01, 0x00, 0xc4])
+      );
+    }
+  }
+
+  public async enableAutoNotifications() {
+    this.logger.log('enabling auto notifications');
+    await this.write(
+      new Uint8Array([0xdf, 0xdf, 0x01, 0x00, 0x01, 0x01, 0xc1])
+    );
+  }
+
+  public async getInt(buffer: Uint8Array) {
+    const bytes = new DataView(new ArrayBuffer(buffer.length));
+    let i = 0;
+    const list = new Uint8Array(bytes.buffer);
+    for (const value of buffer) {
+      list[i] = buffer[i];
+      i++;
+    }
+    return bytes.getInt32(0, false);
+  }
+
+  public override getWeight() {
+    return this.weight.actual;
+  }
+
+  public override getSmoothedWeight() {
+    return this.weight.smoothed;
+  }
+
+  public override getOldSmoothedWeight() {
+    return this.weight.old;
+  }
+
+  private write(_bytes: Uint8Array) {
+    return new Promise((resolve, reject) => {
+      ble.write(
+        this.device_id,
+        DifluidMicrobalance.SERVICE_UUID,
+        DifluidMicrobalance.CHAR_UUID,
+        _bytes.buffer,
+        (e: any) => {
+          this.logger.log(e);
```

Comment:
If e is an object, the logging will log "object", what was the intent of logging the param? If its a valid output which is created, we should JSON.stringify it

Reply:
I removed it.